src/rest/rest_devices.c

```diff
@@ -98,36 +60,155 @@ static int rest_devices_remove_list(linked_list_t *list, const char *id)
     {
         device_data = (database_entry_t *)device_entry->data;
 
-        if (strcmp(id, device_data->uuid) == 0)
+        if (strcmp(uuid, device_data->uuid) == 0)
         {
             linked_list_remove(list, (void *)device_data);
+            database_free_entry(device_data);
             return 0;
         }
     }
 
     return -1;
 }
 
-static json_t *rest_devices_prepare_resp(database_entry_t *device_entry)
+static int append_server_key(json_t *j_object, void *context)
 {
-    json_t *j_resp_obj;
-    char psk_id_buf[512];
-    size_t length = sizeof(psk_id_buf);
+    uint8_t binary_buffer[1024];
+    size_t binary_length;
+    char base64_buffer[1024];
+    size_t base64_length;
+    json_t *j_string;
 
-    if (base64_encode(device_entry->psk_id, device_entry->psk_id_len, psk_id_buf, &length))
+    if (!json_is_object(j_object))
     {
-        return NULL;
+        return -1;
+    }
+
+    binary_length = sizeof(binary_buffer);
+    if (utils_get_server_key(binary_buffer, &binary_length, context))
+    {
+        return -1;
+    }
+
+    base64_length = sizeof(base64_buffer);
+    if (base64_encode(binary_buffer, binary_length, base64_buffer, &base64_length))
+    {
+        return -1;
+    }
+
+    j_string = json_string(base64_buffer);
+    if (j_string == NULL)
+    {
+        return -1;
+    }
+
+    if (json_object_set_new(j_object, "server_key", j_string))
+    {
+        json_decref(j_string);
+        return -1;
     }
 
-    j_resp_obj = json_pack("{s:s, s:s}", "uuid", device_entry->uuid, "psk_id", psk_id_buf);
+    return 0;
+}
+
+static json_t *rest_devices_prepare_resp(database_entry_t *device_entry, void *context)
```

Comment:
I think better name would be something like `rest_devices_entry_to_json` or `rest_devices_entry_to_public_json` if you want to highlight that json format is for sending.